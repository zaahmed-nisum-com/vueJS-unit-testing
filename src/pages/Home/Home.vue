<template>
  <div>
    <input
      type="text"
      id="input_1"
      data-testid="todo-input"
      v-model="newTodo"
    />
    <button data-testid="todo-submit" @click.prevent="addTodo">Add</button>
    <ToDoList :items="items" />
  </div>
</template>

<script>
import ToDoList from "../../components/ToDoList/ToDoList.vue";
export default {
  components: { ToDoList },
  data() {
    return {
      todos: "",
      newTodo: "",
      items: [],
    };
  },
  methods: {
    addTodo() {
      this.todos = this.newTodo;
    },
  },
};
</script>

<style></style>
