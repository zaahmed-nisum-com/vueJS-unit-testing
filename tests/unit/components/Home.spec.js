import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import Home from '../../../src/pages/Home/Home.vue'


describe('Home.vue', () => {

    // test('snapshot', () => {
    //     const wrapper = mount(Home)
    //     expect(wrapper.find('div')).toMatchSnapshot(true)
    // })

    test('allows for adding one todo item', async () => {
        const wrapper = mount(Home)
        wrapper.find('#input_1').setValue('My first todo item')
        // await wrapper.find('[data-testid="todo-submit"]').trigger('click')
        // expect(wrapper.find('[data-testid="todos"]').text()).toContain('My first todo item')
    })
})