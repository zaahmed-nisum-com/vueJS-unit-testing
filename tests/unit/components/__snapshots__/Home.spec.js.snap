// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue snapshot 1`] = `
<div><input type="text" data-testid="todo-input"> <button data-testid="todo-submit">Add</button>
  <div></div>
</div>
`;
